cmd/user-agent/src/pages/chapi/wallet/register/register.js

```diff
@@ -36,6 +36,10 @@ export class RegisterWallet extends WalletManager {
         // create DID
         let did = await this.didManager.createDID(keyType, signType)
 
+        // Make sure at least the username is stored so that the DID can be saved to SDS
+        // This will be overridden by another call at the end of the function that contains the full wallet metadata
+        await this.storeWalletMetadata(user, {})
```

Comment:
saving jsindex db twice for saving same record in single function is a performance bottleneck, add a followup TODO to avoid this. May be a better flow would save a db call here.

Reply:
Added a follow-up TODO. Perhaps when we add the proper login flow there will be a solution to this... I'll keep an eye out.